### ISSUE TITLE
Rregarding model fine-tuning

### ISSUE THREAD

**Issue by oleksost** [akoksal/LongForm#3]
Is it possible to have a look at model fine-tuning code for the models mentioned in the paper? Thank you!

Would be specifically interested in the hyperparameters used for fine-tuning as well as how you handle the long sequences in the fine-tuning process? Are you concatenating all instructions and outputs together and split the resulting text into chunks? Are you calculating the loss for the instruction part of the input as well as for the output or only for the output (like in Alpaca)?

**Comment by akoksal** [comment]
I use generic autoregressive model finetuning from HuggingFace trainer with the following tricks:
- Group texts with [this function](https://huggingface.co/docs/transformers/tasks/language_modeling) with the block size of 1024.
- Include end-of-instruction ([EOI]) token at the end of the input/instruction to clarify the separation between input and output.
- Calcluate the loss for the whole part: input and output.
- Most importantly I'm using the following DeepSpeed config for the optimizations and set it in the HuggingFace trainer (I enabled bf16 optimization in the HuggingFace trainer):
```
{
    "fp16": {
        "enabled": "auto",
        "loss_scale": 0,
        "loss_scale_window": 1000,
        "initial_scale_power": 16,
        "hysteresis": 2,
        "min_loss_scale": 1
    },

    "bf16": {
        "enabled": "auto"
    },

    "optimizer": {
        "type": "AdamW",
        "params": {
            "lr": "auto",
            "betas": "auto",
            "eps": "auto",
            "weight_decay": "auto"
        }
    },

    "scheduler": {
        "type": "WarmupLR",
        "params": {
            "warmup_min_lr": "auto",
            "warmup_max_lr": "auto",
            "warmup_num_steps": "auto"
        }
    },

    "zero_optimization": {
        "stage": 3,
        "offload_optimizer": {
            "device": "cpu",
            "pin_memory": true
        },
        "allgather_bucket_size": 2e8,
        "overlap_comm": true,
        "contiguous_gradients": true,
        "sub_group_size": 1e12,
        "reduce_bucket_size": 2e8,
        "stage3_prefetch_bucket_size": "auto",
        "stage3_param_persistence_threshold": "auto",
        "stage3_max_live_parameters": 1e9,
        "stage3_max_reuse_distance": 1e9,
        "stage3_gather_16bit_weights_on_model_save": true
    },
    
    "gradient_accumulation_steps": "auto",
    "gradient_clipping": "auto",
    "steps_per_print": 50,
    "train_batch_size": "auto",
    "train_micro_batch_size_per_gpu": "auto",
    "wall_clock_breakdown": false
}
```

Hope it would help!
